coders/dcm.c

```diff
@@ -2877,32 +2880,43 @@ static MagickBooleanType ReadDCMPixels(Image *image,DCMInfo *info,
                   }
                 i++;
               }
-          index=(int) ((pixel_value*info->rescale_slope)+
-            info->rescale_intercept);
-          if (info->window_width == 0)
+          if (info->rescale)
             {
-              if (info->signed_data == 1)
-                index-=32767;
+              double
+                scaled_value;
+
+              scaled_value=pixel_value*info->rescale_slope+
+                info->rescale_intercept;
+              if (info->window_width == 0)
+                {
+                  index=(int) scaled_value;
+                }
+              else
+                {
+                  double
+                    window_max,
+                    window_min;
+    
+                  window_min=ceil(info->window_center-
+                    (info->window_width-1.0)/2.0-0.5);
+                  window_max=floor(info->window_center+
+                    (info->window_width-1.0)/2.0+0.5);
+                  if (scaled_value <= window_min)
+                    index=0;
+                  else
+                    if (scaled_value > window_max)
+                      index=(int) info->max_value;
+                    else
+                      index=(int) (info->max_value*(((scaled_value-
+                        info->window_center-0.5)/(info->window_width-1))+0.5));
+                }
             }
           else
             {
-              ssize_t
-                window_max,
-                window_min;
-
-              window_min=(ssize_t) ceil((double) info->window_center-
-                (info->window_width-1.0)/2.0-0.5);
-              window_max=(ssize_t) floor((double) info->window_center+
-                (info->window_width-1.0)/2.0+0.5);
-              if ((ssize_t) index <= window_min)
-                index=0;
-              else
-                if ((ssize_t) index > window_max)
-                  index=(int) info->max_value;
-                else
-                  index=(int) (info->max_value*(((index-info->window_center-
-                    0.5)/(info->window_width-1))+0.5));
+              index=pixel_value;
             }
+          if (info->signed_data == 1)
+            index-=32767;
```

Comment:
Yes, it seemed correct to me. If the output is signed, you must offset the origin, whether or not you interpret the window. I have not test files I could check with, though.